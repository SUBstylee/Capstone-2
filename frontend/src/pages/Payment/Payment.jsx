import { useState } from "react";
import { useNavigate } from "react-router-dom";
import {Form,Button,Col, Row} from 'react-bootstrap';
import { useDispatch,useSelector } from "react-redux";
import { savePaymentMethod } from "../../actions/cartActions";
import FormContainer from "../../components/FormContainer/FormContainer";
import CheckoutSteps from "../../components/CheckoutSteps/CheckoutSteps";
import MetaWrapper from "../../components/MetaWrapper/MetaWrapper";

import './Payment.scss'

const Payment = () => {
    const cart=useSelector(state=>state.cart);
    const {shippingAddress}=cart;
    const dispatch=useDispatch();
    const navigate=useNavigate();

    if(!shippingAddress)navigate('/shipping');

    const [paymentMethod,setPaymentMethod]=useState('PayPal');

    const submitHandler=(e)=>{
        e.preventDefault();
        dispatch(savePaymentMethod(paymentMethod));
        navigate('/placeorder');
    };

    return (
        <div className="Payment">
            <MetaWrapper title='TAA-Payment'/>
            <FormContainer>
                <CheckoutSteps step1 step2 step3/>
                <h1>Payment Method</h1>
                <Form onSubmit={submitHandler}>
                    <Form.Group>
                        <Form.Label as='legend'>Select Method</Form.Label>
                        <Row>
                        <Col className="payment-icons" xs={1}>
                            <label htmlFor="PayPal"><i className="fa-brands fa-paypal"></i></label><br/>
                            <label htmlFor="Stripe"><i className="fa-brands fa-stripe-s"></i></label>
                        </Col>
                        <Col xs={2}>
                            <Form.Check type='radio' label='PayPal' id='PayPal' name='paymentMethod' value='PayPal' checked onChange={(e)=>setPaymentMethod(e.target.value)}></Form.Check>
                            <Form.Check type='radio' label='Stripe' id='Stripe' name='paymentMethod' value='Stripe' onChange={(e)=>setPaymentMethod(e.target.value)}></Form.Check>
                        </Col>
                        </Row>
                    </Form.Group>
                    <Button className='mt-2' type='submit' variant="primary">Continue</Button>
                </Form>
            </FormContainer>   
        </div>
    );
};

export default Payment;