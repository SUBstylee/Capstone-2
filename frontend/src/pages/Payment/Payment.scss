.Payment{
    .payment-icons{
        margin-right: -25px;
    }
}