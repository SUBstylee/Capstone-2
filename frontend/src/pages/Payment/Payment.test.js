import { render } from '@testing-library/react';
import Payment from './Payment.jsx';
import store from '../../store';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

test("renders Payment component without crashing", () => {
    const { getByText } = render(
        <Provider store={store}><BrowserRouter><Payment /></BrowserRouter></Provider>
    );

    expect(getByText("PayPal")).toBeInTheDocument();
});

test("matches snapshot", function () {
    const { asFragment } = render(
        <Provider store={store}><BrowserRouter><Payment /></BrowserRouter></Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});