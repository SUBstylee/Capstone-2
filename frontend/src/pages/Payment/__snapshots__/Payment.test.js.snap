// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Payment"
  >
    <div
      class="FormContainer"
    >
      <div
        class="container"
      >
        <div
          class="justify-content-md-center row"
        >
          <div
            class="col-md-6 col-12"
          >
            <div
              class="CheckoutSteps"
            >
              <div
                class="justify-content-center mb-4 nav"
              >
                <div
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    data-rr-ui-event-key="/login"
                    href="/login"
                  >
                    Sign In
                  </a>
                </div>
                <div
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    data-rr-ui-event-key="/shipping"
                    href="/shipping"
                  >
                    Shipping
                  </a>
                </div>
                <div
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    data-rr-ui-event-key="/payment"
                    href="/payment"
                  >
                    Payment
                  </a>
                </div>
                <div
                  class="nav-item"
                >
                  <a
                    aria-disabled="true"
                    class="nav-link disabled"
                    role="button"
                  >
                    Place Order
                  </a>
                </div>
              </div>
            </div>
            <h1>
              Payment Method
            </h1>
            <form
              class=""
            >
              <div>
                <legend
                  class="form-label"
                >
                  Select Method
                </legend>
                <div
                  class="row"
                >
                  <div
                    class="payment-icons col-1"
                  >
                    <label
                      for="PayPal"
                    >
                      <i
                        class="fa-brands fa-paypal"
                      />
                    </label>
                    <br />
                    <label
                      for="Stripe"
                    >
                      <i
                        class="fa-brands fa-stripe-s"
                      />
                    </label>
                  </div>
                  <div
                    class="col-2"
                  >
                    <div
                      class="form-check"
                    >
                      <input
                        checked=""
                        class="form-check-input"
                        id="PayPal"
                        name="paymentMethod"
                        type="radio"
                        value="PayPal"
                      />
                      <label
                        class="form-check-label"
                        for="PayPal"
                        title=""
                      >
                        PayPal
                      </label>
                    </div>
                    <div
                      class="form-check"
                    >
                      <input
                        class="form-check-input"
                        disabled=""
                        id="Stripe"
                        name="paymentMethod"
                        type="radio"
                        value="Stripe"
                      />
                      <label
                        class="form-check-label"
                        for="Stripe"
                        title=""
                      >
                        Stripe
                      </label>
                    </div>
                  </div>
                </div>
              </div>
              <button
                class="mt-2 btn btn-primary"
                type="submit"
              >
                Continue
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
